models/user.rb

```diff
@@ -61,12 +61,17 @@ def to_hash(params={})
           ).count
 
           # Note that the comments may have been responses to a thread not started by author_id.
-          comment_thread_ids = Comment.where(
+
+          # We need to include the comment_thread in thie query to prevent an N+1 query
+          # when we iterate through all of the comments to get their thread context
+          comment_thread_ids = Comment.includes(:comment_thread).where(
```

Comment:
The short version is that `c.context` isn't a straight property access, but instead delegates to the comment's thread. So even though nothing looks like a thread is used, the object will be accessed.